src/guardianSigner.ts

```diff
@@ -0,0 +1,47 @@
+import { ISignable, ISignature } from "./interface";
+import { UserSecretKey } from "./userKeys";
+import { UserSigner } from "./userSigner";
+import { Signature } from "./signature";
+import { ErrSignerCannotSign } from "./errors";
+
+/**
+ * ed25519 signer
+ */
+export class GuardianSigner extends UserSigner {
+    private readonly guardianSecretKey: UserSecretKey;
+
+    constructor(guardianSecretKey: UserSecretKey) {
+        super(guardianSecretKey)
+        this.guardianSecretKey = guardianSecretKey
+    }
+
+/**
+ * Signs a message.
+ * @param signable the message to be signed (e.g. a {@link Transaction}).
+ */
+    async guard(signable: ISignable): Promise<void> {
+        try {
+            this.tryGuard(signable);
+        } catch (err: any) {
+            throw new ErrSignerCannotSign(err);
+        }
+    }
+
+    private tryGuard(signable: ISignable) {
+        let ownerSignature = signable.getSignature()
```

Comment:
you can change the variable declaration to all the variables that are not modified from `let` to `const`